_posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md

```diff
@@ -0,0 +1,184 @@
+ 
+
+### How the ultrasonic sensor works
+
+The HC-SR04 device emits an ultrasound at 40kHz which travels through the air and if there is an object or obstacle on its path, the sound will bounce back to the module. 
+Considering the travel time and the speed of the sound is 343m/s the object distance from the sensor can be calculated.
+
+The ultrasonic sensor uses sonar to determine the distance to an object. Here’s what happens:
+1. The transmitter (trig pin) sends a signal: a high-frequency sound.
+2. When the signal finds an object, it is reflected 
+3. The transmitter (echo pin) receives it.
+
+<img class="img-fluid" src="{{'assets/posts/2019-09-20-arduino-ultrasonic-sensor/ultrasonic-sensors-operation.png' | relative_url}}"/>
+
+* The time between the transmission and reception of the signal allows us to calculate the distance to an object. 
+* This is possible because we know the sound’s velocity in the air.
+
+
+
+### Calculation
+
+```text
+  
+  Speed of sound, v                = 343/s 
+  Expressed in cm per micro second = 0.0343 cm/µs
+ 
+  Distance = Speed / Time
+  Time = Distance / speed
+      
+  e.g. Object at 10cm away
+  
+  Time = 10 / 0.0343 =  294 µ s (this needs to be halved to cater for the sound performs a round trip - forward and backward) 
+  
+  Distance = Time * 0.0343 / 2
+
+```
+
+   For example, if the object is 10 cm away from the sensor, and the speed of the sound is 343 m/s or 0.0343 cm/µs the sound wave will need to travel about 294 µ seconds. 
+   But what you will get from the Echo pin will be double that number because the sound wave needs to travel forward and bounce backward.  So in order to get the distance in cm we need to multiply the received travel time value from the echo pin by 0.0343 and divide it by 2.
+
+
+#### HC-SR04 Timing diagram
+
+* The circuit needs to supply a short 10uS pulse to the trigger input to start the ranging, and then the module will send out
+an 8 cycle burst of ultrasound at 40 kHz and raise its echo. 
+* The Echo is avdistance object that is pulse width and the range in proportion .You can
+calculate the range through the time interval between sending trigger signal and
+receiving echo signal. Formula: uS / 58 = centimeters or uS / 148 =inch; or: the
+range = high level time * velocity (340M/S) / 2; we suggest to use over 60ms
+measurement cycle, in order to prevent trigger signal to the echo signal. 
+
+<img class="img-fluid" src="{{'assets/posts/2019-09-20-arduino-ultrasonic-sensor/ultrasonic-timing-diagram.png' | relative_url}}"/>
+
+
+
+### Component diagram
+
+The following shows the a breadboard component diagram which will be used for this project.
+
+<img class="img-fluid" src="{{'assets/posts/2019-09-20-arduino-ultrasonic-sensor/ultrasonic-range-sensor.png' | relative_url}}"/>
+
+
+### Code
+
+* The following code snippet can be used  (note it's partially incomplete - you can complete yourself)
+
+```
```

Comment:
The syntax highlighter supports `cpp`.
See https://github.com/rouge-ruby/rouge/wiki/List-of-supported-languages-and-lexers